pkg/segment/results/mresults/metricresults.go

```diff
@@ -398,11 +398,15 @@ func (res *MetricsResult) GetMetricTagsResultSet(mQuery *structs.MetricsQuery) (
 			return nil, nil, err
 		}
 		for index, val := range tagValues[:len(tagValues)-1] {
-			tagKeyValueSet = append(tagKeyValueSet, fmt.Sprintf("%s:%s", uniqueTagKeys[index], val))
+			tagKeyValueSet[fmt.Sprintf("%s:%s", uniqueTagKeys[index], val)] = true
 		}
 	}
+	uniqueTagKeysSet := make([]string, 0)
+	for key := range tagKeyValueSet {
+		uniqueTagKeysSet = append(uniqueTagKeysSet, key)
+	}
```

Comment:
Might want to move this into the utils package if you think we'll convert sets to slices in other places